configuration/config2/CfgStringLimits.hpp

```diff
@@ -0,0 +1,44 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _CONFIG2_CFGSTRINGLIMITS_HPP_
+#define _CONFIG2_CFGSTRINGLIMITS_HPP_
+
+#include "CfgLimits.hpp"
+
+
+class CfgStringLimits : public CfgLimits
+{
+    public:
+
+        CfgStringLimits() : m_removeWhiteSpace(true) { };
+        virtual ~CfgStringLimits() { };
+        void setRemoveWhiteSpace(bool remove) { m_removeWhiteSpace = true; }
```

Comment:
I was not quite sure where to put this. There is a string restriction to collapse (remove) all whitespace from a value. The restriction is for the XML processor (I believe when the schema is used to validate XML, this tells the processor to remove the whitespace), but since we don't have one that is used, we needed to capture it, just in case. Since the CfgType is a lightweight class and all of the actual type checking is done by the Limits object, I placed it there. While not using it yet, the intent was that when a value was checked for validity through the Limits object, if this flag was set, whitespace would be removed prior to saving the value to the environment.

<xs:restriction base="xs:string">
      <xs:whiteSpace value="collapse"/>
</xs:restriction>